src/pages/groupComparison/GroupComparisonMutationMapper.tsx

```diff
@@ -46,14 +50,127 @@ export default class GroupComparisonMutationMapper extends MutationMapper<
     }
 
     /**
-     * Override the parent method to get custom controls.
+     * Overriding the parent method to have a customized filter panel.
      */
-    protected get customControls(): JSX.Element | undefined {
+    protected get mutationFilterPanel(): JSX.Element | null {
         return (
-            <PercentToggle
-                axisMode={this.props.axisMode}
-                onScaleToggle={this.props.onScaleToggle}
-            />
+            <div>
+                {this.props.isPutativeDriver ? (
+                    <div
+                        style={{
+                            paddingTop: 5,
+                            paddingBottom: 5,
+                        }}
+                    >
+                        <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                            (A){' '}
+                            {
+                                this.store.dataStore
+                                    .sortedFilteredGroupedData[0].group
+                            }
+                        </div>
+                        <DriverAnnotationProteinImpactTypeBadgeSelector
+                            filter={this.proteinImpactTypeFilter}
+                            counts={this.mutationCountsByProteinImpactTypeForGroup(
+                                0
+                            )}
+                            onSelect={this.onProteinImpactTypeSelect}
+                            onClickSettingMenu={this.props.onClickSettingMenu}
+                            annotatedProteinImpactTypeFilter={
+                                this.annotatedProteinImpactTypeFilter
+                            }
+                        />
```

Comment:
This should probably be a separate function or component which can also be used for the mirrored part to avoid duplication. I think the parameters should be the `group index` and the `short name`.